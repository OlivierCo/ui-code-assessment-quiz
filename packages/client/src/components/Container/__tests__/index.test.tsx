import React from 'react'
import { mount, shallow } from 'enzyme'
import { Container } from '../index'

describe('Container', () => {
  it('should be defined', () => {
    expect(Container).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = shallow(
      <Container center scrollable>
        <div className="foo">Bar</div>
      </Container>,
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should render Container with children', () => {
    const wrapper = mount(
      <Container center scrollable>
        <div className="foo">Bar</div>
      </Container>,
    )

    expect(wrapper.contains(<div className="foo">Bar</div>)).toBe(true)
    expect(wrapper.prop('center')).toEqual(true)
    expect(wrapper.prop('scrollable')).toEqual(true)
  })
})
