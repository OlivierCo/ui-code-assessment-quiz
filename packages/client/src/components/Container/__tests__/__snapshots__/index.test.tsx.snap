// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render correctly 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Container"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  css={
    Object {
      "map": undefined,
      "name": "kcztdl",
      "next": undefined,
      "styles": "position:absolute;top:0;bottom:0;right:0;left:0;left:50%;top:50%;transform:translate(-50%, -50%);@media (max-width: 420px){width:100%;};overflow-y:scroll;;",
      "toString": [Function],
    }
  }
>
  <div
    className="foo"
  >
    Bar
  </div>
</EmotionCssPropInternal>
`;
